src/domain/repositories/DataValueRepository.ts

```diff
@@ -8,9 +9,9 @@ export interface DataValueRepository {
 
 export interface GetDataValueOptions {
     orgUnitIds: Id[];
-    dataElementGroupIds?: Id[];
     children: boolean;
     includeDeleted: boolean;
     startDate: string;
     endDate: string;
+    tempDataElementGroup: Maybe<{ code: Code; dataElements: Ref[] }>;
```

Comment:
The fact that we need to create a temporal dataElementGroup to efficiently query the dataValues endpoint is an (unfortunate) implementation detail. The domain should know nothing about it: confine any related code to the data repo.